### DISCUSSION TITLE
New icons for data types with text icons

### DISCUSSION THREAD

**Discussion by snw-0** [godotengine/godot-proposals#4455] [Editor]
Many data types have icons that are just their name (or an abbreviation of it) in text:

![icons](https://user-images.githubusercontent.com/2808851/165592010-25341b34-3120-49e7-a82f-5a6fb65edba4.png)

It could be worth making some new, more symbolic icons for some or all of these, along the lines of the icons used for node types.  I can create the SVG files.

We discussed it a bit on Discord and here are some possibilities (thanks to Lyuma and eons for contributing many of these):
- AABB: a cube or rectangular prism
- Array: "[ ]"
- Basis: a stylized 3x3 matrix, or the three-arrows symbol for a basis
- bool: "0/1", or a radio button, or two circles with one filled in
- Color: a square or circle with red, green, and blue stripes, as used in e.g. ColorRect; or three small circles colored RGB
- Dictionary: "{ }"
- Enum: perhaps left as is
- float: a particular float, e.g. "1.0" or "π"; or a blackboard-bold R (i.e. the field of real numbers)
- int: a particular int, e.g. "1"; or a blackboard-bold Z (i.e. the ring of intgers)
- MiniObject: I'm not sure where this is used, but it could use the box shape of Object.svg
- Nil: "∅", or left as is
- NopePath: "//", or "$", or an arrow pointing to the circle used for nodes
- Plane: the rhombus used for PlaneMeshes in a different color
- Quaternion: "Q", or a blackboard-bold H (i.e. the algebra of quaternions), or some sort of stylized rotation arrow
- Rect2: a rectangle in the blue color used for 2D nodes
- Rect2i: as Rect2 but with white corners
- RID: perhaps left as is, or just "ID"
- String: "A", or left as is (since "Str" is itself a string)
- StringName: as String but underlined or in a different color
- Transform2D: perhaps a representation of some shape being transformed
- Transform3D: as Transform2D but in red for 3D
- UV: probably left as is
- Variant: an asterisk or question mark
- Vector2: an arrow, in blue for 2D
- Vector2i: as Vector2 but with a white dot at the head
- Vector3, Vector 3i: as Vector2 but in red for 3D
- MemberConstant, MemberMethod, MemberProperty: perhaps left as is
- Packed___Array: the existing design of putting brackets around the other type is fine

Please feel free to suggest alternatives.

**Comment by goatchurchprime** [comment]
Nice idea.  Unfortunately for this one the science is in: https://uxmyths.com/post/715009009/myth-icons-enhance-usability

**Reply by Calinou** [reply]
> Nice idea. Unfortunately for this one the science is in: [uxmyths.com/post/715009009/myth-icons-enhance-usability](https://uxmyths.com/post/715009009/myth-icons-enhance-usability)

I wouldn't remove the type icons. I still find them useful for identification at a glance, especially thanks to their color (since the text labels are not colored, and it's not always easy to do).

Godot doesn't use type icons without having corresponding text next to them. Therefore, the usability caveat mentioned in that post doesn't apply here.
